### PR TITLE
#6Task.alex.moiseyenko 

### PR DESCRIPTION
**Pull request by alexmoiseyenko**
https://travis-ci.org/github/alexmoiseyenko/js-assignments/builds/707104910

### CODE REVIEW

**Comment by KostyaKov** [comment]
Hello. 
In Travis test there are no functions:
- isBracketsBalanced should check the balanced brackets
- getCommonDirectoryPath should return the n-ary string representation of number
- evaluateTicTacToePosition should return the winner if defined
Can you add them, please?

**Comment by KostyaKov** [comment]
Can you provide new Travis link (with added functions), please?

**Comment by alexmoiseyenko** [comment]
https://travis-ci.org/github/alexmoiseyenko/js-assignments/builds/722773768

**Comment by alexmoiseyenko** [comment]
> Can you provide new Travis link (with added functions), please?

For sure, I didn't know it was necessary

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -30,7 +30,14 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+    let res = '';
+    if (num%3 === 0) {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: it is necessary to observe indents

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -82,75 +96,89 @@ function getSumBetweenNumbers(n1, n2) {
  *   10,10,10 =>  true
  */
 function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+    if ((a + b > c) && (b + c > a) && (a + c > b)) {
+        return true;
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: You can return expression result without "if"

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -82,75 +96,89 @@ function getSumBetweenNumbers(n1, n2) {
...
  *   { top: 0, left: 0, width: 10, height: 10 },
  *   { top:20, left:20, width: 20, height: 20 }    =>  false
- *  
+ *
  */
 function doRectanglesOverlap(rect1, rect2) {
-    throw new Error('Not implemented');
+    let top1 = rect1.top + rect1.height;
+    let left1 = rect1.left + rect1.width;
+    if (rect2.top < top1 && rect2.left < left1) {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: You can return expression result without "if"

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -192,7 +237,19 @@ function findFirstSingleChar(str) {
...
 function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
-    throw new Error('Not implemented');
+    if (b < a) {
+        let c = a;
+        a = b;
+        b = c;
+    }
+    if (isStartIncluded && isEndIncluded) {
+        return `[${a}, ${b}]`;
+    } else if (!isStartIncluded && isEndIncluded) {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: "else" is redundant after "return"

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -332,7 +439,35 @@ function isBracketsBalanced(str) {
...
-    throw new Error('Not implemented');
+    let secondsDiff = (endDate - startDate) / 1000;
+    let minutesDiff = Math.round((secondsDiff - 0.001) / 60);
+    let hoursDiff = Math.round((secondsDiff - 0.001) / 60 / 60);
+    let daysDiff = Math.round((secondsDiff - 0.001) / 24 / 60 / 60);
+    let monthsDiff = Math.round(daysDiff / 30);
+    let yearsDiff = Math.round(daysDiff / 365);
+    if (secondsDiff <= 45) {
+        return `a few seconds ago`;
+    } else if (secondsDiff <= 90) {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: "else" is redundant after "return"

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by alexmoiseyenko** [comment]
https://travis-ci.org/github/alexmoiseyenko/js-assignments/builds/723488648

**Review by KostyaKov — COMMENTED** [review]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**